.github/workflows/release.yml

```diff
@@ -0,0 +1,88 @@
+name: Auto-release on PR merge
+
+on:
+  # ATM, this is the closest trigger to a PR merging
+  push:
+    branches:
+      - master
+
+jobs:
+  auto-release:
+    runs-on: ubuntu-latest
+    if: "!contains(github.event.head_commit.message, 'ci skip') && !contains(github.event.head_commit.message, 'skip ci')"
+    outputs:
+      auto-version: ${{ steps.auto-version.outputs.version }}
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Download auto
+        run: |
+          #curl -vL -o - "$(curl -fsSL https://api.github.com/repos/intuit/auto/releases/latest | jq -r '.assets[] | select(.name == "auto-linux.gz") | .browser_download_url')" | gunzip > ~/auto
+          # Pin to 10.16.1 so we don't break if & when
+          # <https://github.com/intuit/auto/issues/1778> is fixed.
+          wget -O- https://github.com/intuit/auto/releases/download/v10.16.1/auto-linux.gz | gunzip > ~/auto
+          chmod a+x ~/auto
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '^3.7'
+
+      - name: Install Python dependencies
+        run: python -m pip install build bump2version twine
+
+      - name: Check whether a release is due
+        id: auto-version
+        run: |
+          version="$(~/auto version)"
+          echo "::set-output name=version::$version"
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Create release
+        run: ~/auto shipit
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          TWINE_USERNAME: __token__
+          TWINE_PASSWORD: ${{ secrets.PYPI_TOKEN }}
+
+  build-docker:
+    runs-on: ubuntu-latest
+    needs: auto-release
+    if: needs.auto-release.outputs.auto-version != ''
+    steps:
+      - name: Get tag of latest release
+        id: latest-release
+        run: |
+          latest_tag="$(curl -fsSL https://api.github.com/repos/$GITHUB_REPOSITORY/releases/latest | jq -r .tag_name)"
+          echo "::set-output name=tag::$latest_tag"
```

Comment:
why `needs.auto-release.outputs.auto-version` can't be used here? I am afraid if api.github has some delay in propagating effects of `auto shipit`  we might end up using some wrong tag here.

Reply:
Because `auto version` outputs a patch level, not a version number.

Reply:
I feel that we had similar problem somewhere else and did find how to trick `auto` to tell us the full version, didn't we?  I would really prefer to not rely on the logic of jumping through github without any validation that the tag does correspond to this state. Wouldn't prior step tag it, so we could just use `git describe` here and get a clean version?

Reply:
Workflow adjusted.